internal/worker/sshserver/k8s/session_test.go

```diff
@@ -79,3 +87,88 @@ func (s *k8sSessionSuite) TestSessionHandler(c *gc.C) {
 	k8sHandlers.SessionHandler(s.session)
 	c.Assert(readWriter.String(), gc.Equals, "failed to resolve k8s exec info: error\n")
 }
+
+func (s *k8sSessionSuite) TestSessionHandlerPty(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+	l := loggo.GetLogger("test")
+	s.resolver.EXPECT().ResolveK8sExecInfo(gomock.Any()).Return(params.SSHK8sExecResult{
+		PodName:   "test-pod",
+		Namespace: "test-namespace",
+	}, nil)
+	s.executor.EXPECT().Exec(gomock.Any(), gomock.Any()).DoAndReturn(func(params k8sexec.ExecParams, opts <-chan struct{}) error {
+		c.Assert(params.PodName, gc.Equals, "test-pod")
+		c.Assert(params.ContainerName, gc.Equals, "test-container")
+
+		// Simulate a write
+		_, err := params.Stdout.Write([]byte("test output"))
+		c.Assert(err, jc.ErrorIsNil)
+		return nil
+	})
+
+	virtualHostame, err := virtualhostname.NewInfoContainerTarget(uuid.New().String(), "test/0", "test-container")
+	c.Assert(err, jc.ErrorIsNil)
+
+	k8sHandlers, err := NewHandlers(
+		virtualHostame,
+		s.resolver,
+		l,
+		func(string) (k8sexec.Executor, error) {
+			return s.executor, nil
+		},
+	)
+	c.Assert(err, jc.ErrorIsNil)
+	s.session.EXPECT().Environ()
+	s.session.EXPECT().Command()
+	s.context.EXPECT().Done()
+	s.session.EXPECT().Context().Return(s.context)
+	mockSession := userSession{
+		Session: s.session,
+		isPty:   true,
+	}
+	// Simulate the user closing the session.
+	go func() {
+		time.Sleep(testing.ShortWait)
+		mockSession.Close()
+	}()
```

Comment:
I think you can and should refactor this to not be reliant on a time.Sleep().
The easiest thing is to have the target close its end of the connection after writing "test output". Then you don't need to close the client, you can just run `k8sHandlers.SessionHandler(&mockSession)` and it should return.